pkgs/misc/emulators/wine/base.nix

```
@@ -94,6 +94,15 @@ stdenv.mkDerivation ((lib.optionalAttrs (buildScript != null) {
   # and enable doCheck
   doCheck = false;
 
+  # the relevant check in configure.ac will fail under nix as we are not
+  # using the system supplied compiler tools. We will subsequently *not*
+  # pass `-mmacosx-version-min` in the LDFLAGS, and end up with a linked
+  # binary against 10.14 which *does* contain a __DATA,__dyld section and
+  # then horribly fail to run on Mojave.
+  postConfigure = lib.optional (stdenv.hostPlatform.isDarwin) ''
```

This should work as well:
```suggestion
  makeFlags = lib.optional (stdenv.hostPlatform.isDarwin) "LDFLAGS=-mmacosx-version-min=10.7";
```
Otherwise replace `optional` with `optionalString` and please use substituteInPlace instead of sed as it prints warning if the original string is not found.